DREAM/tasdreamExternalTests.cpp

```diff
@@ -38,7 +38,18 @@ using std::cerr;
 using std::endl;
 using std::setw;
 
-ExternalTester::ExternalTester(int num_monte_carlo) : num_mc(num_monte_carlo){
+TestRNG::TestRNG(int seed): s(seed % 2097165){}
+TestRNG::~TestRNG(){}
+
+double TestRNG::getSample01() const{
+    s *= 511;
+    s += 127;
+    s %= 2097165;
+    return (double)(((double) s) / 2097165.0);
+}
```

Comment:
This is the linear congruential algorithm isn't it?  Why don't you use what is implemented in the standard header `<random>`?

Reply:
C.f. http://en.cppreference.com/w/cpp/numeric/random

Reply:
Yes, this is a rather basic algorithm, but it is sufficient. I was using <random>, but that was the main problem with clang5, and then happened again when I was using Visual C++. GCC, different versions of Clang (e.g., 3.8, vs 4.0 vs 5.0), and MS Visual Studio, all use different random number generators, meaning that the same seed gives different sequences of samples for the different compilers. I'm guessing ICC and PGI will be also different. DREAM is a random sampler and tests have to be done statistically, e.g., with 95% or 99% confidence. I have 5 - 6 tests, if each passes at the 95% then I have 25% failure rate based on the random seed. Due to the differences in compilers, I cannot simply fix the seed ... with a custom compiler, I don't have to worry about that.

This is done only for the tests, by default, the library uses <random> and the user can opt to specify their own random number generator.

Reply:
```C++
// I would just use the standard minstd_rand engine.
// But you can always replace with `std::linear_congruential_engine<std::uint_fast32_t, 512, 127, 2097165>` if you like.
// Pass a seed to the constructor if you don't like the default.
std::minstd_rand generator();

// Then I assume you were trying to get a uniform distribution over [-1, 1]
std::uniform_real_distribution<double> distribution(-1., 1.);

// Then do
for (int i; ...)
  sample[i] = distribution(generator);  
```

Reply:
I see your point, C++11 does have standard that should be compiler independent. However, using this won't save on code unless I also change the way the core distribution is assigned to the DREAM and CorePDF classes. It is a design decision to consider, I don't want to block the user from specifying something potentially more exotic than what is in the C++11 standard.